magento-research/pwa-studio
# Peregrine: Refactoring image carousel component for peregrine

## Description

This PR contains product page image gallery code refactored for Peregrine.

Noteable Changes:
* State/API around Carousel logic has been moved to a hook.
* A new `Image` component has been added to abstract the placeholder rendering functionality. 
* Replaced `List` usage with inline map in `ThumbnailList` to prevent unnecessary re-renders.

Closes #1319.
Also, if the `Image` component in this PR makes it in this also closes #1231 

## Verification Steps

1. Go to the Product page.
2. Verify the Image gallery shows up.
3. Make sure Image gallery works as expected.

## How Have YOU Tested this?



Manual testing on product page image gallery

## Screenshots / Screen Captures (if appropriate):

## Proposed Labels for Change Type/Package




## Checklist:
